tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -309,20 +398,35 @@ ROOT::Experimental::Detail::RPage ROOT::Experimental::Detail::RPageSourceFile::P
       sealedPageBuffer = onDiskPage->GetAddress();
    }
 
-   std::unique_ptr<unsigned char []> pageBuffer;
+   const auto nElts = pageInfo.fNElements;
+   auto newPage = fPageAllocator->NewPage(columnId, elementSize, nElts);
    {
       RNTupleAtomicTimer timer(fCounters->fTimeWallUnzip, fCounters->fTimeCpuUnzip);
-      pageBuffer = UnsealPage({sealedPageBuffer, bytesOnStorage, pageInfo.fNElements}, *element);
+      const auto bytesPacked = element->GetPackedSize(nElts);
+      if (bytesOnStorage != bytesPacked) {
+         fDecompressor->Unzip(sealedPageBuffer, bytesOnStorage, bytesPacked, newPage.GetBuffer());
+      } else {
+         // We cannot simply map the sealed page as we don't know its life time. Specialized page sources
+         // may decide to implement to not use UnsealPage but to custom mapping / decompression code.
+         // Note that usually pages are compressed.
+         memcpy(newPage.GetBuffer(), sealedPageBuffer, bytesPacked);
+      }
+      if (!element->IsMappable()) {
+         auto tmp = fPageAllocator->NewPage(columnId, elementSize, nElts);
+         element->Unpack(tmp.GetBuffer(), newPage.GetBuffer(), nElts);
+         fPageAllocator->DeletePage(newPage);
+         newPage = tmp;
+      }
```

Comment:
We should reuse the logic in `UnsealPage()`.  Instead, its implementation should be probable pagecache-aware.
We'll see how to do this after the vacation period.